backend/open_webui/socket/main.py

```diff
@@ -54,64 +54,108 @@
 
 if WEBSOCKET_MANAGER == "redis":
     log.debug("Using Redis to manage websockets.")
-    SESSION_POOL = RedisDict("open-webui:session_pool", redis_url=WEBSOCKET_REDIS_URL)
-    USER_POOL = RedisDict("open-webui:user_pool", redis_url=WEBSOCKET_REDIS_URL)
-    USAGE_POOL = RedisDict("open-webui:usage_pool", redis_url=WEBSOCKET_REDIS_URL)
-
-    clean_up_lock = RedisLock(
-        redis_url=WEBSOCKET_REDIS_URL,
-        lock_name="usage_cleanup_lock",
-        timeout_secs=TIMEOUT_DURATION * 2,
-    )
-    aquire_func = clean_up_lock.aquire_lock
-    renew_func = clean_up_lock.renew_lock
-    release_func = clean_up_lock.release_lock
+    try:
+        SESSION_POOL = RedisDict(
+            "open-webui:session_pool", redis_url=WEBSOCKET_REDIS_URL
+        )
+        USER_POOL = RedisDict("open-webui:user_pool", redis_url=WEBSOCKET_REDIS_URL)
+        USAGE_POOL = RedisDict("open-webui:usage_pool", redis_url=WEBSOCKET_REDIS_URL)
+
+        clean_up_lock = RedisLock(
+            redis_url=WEBSOCKET_REDIS_URL,
+            lock_name="usage_cleanup_lock",
+            timeout_secs=TIMEOUT_DURATION * 2,
+        )
+        acquire_func = clean_up_lock.acquire_lock
+        renew_func = clean_up_lock.renew_lock
+        release_func = clean_up_lock.release_lock
+    except Exception as e:
+        log.error(
+            f"Failed to initialize Redis websocket manager: {e}. Falling back to local manager."
+        )
+        # Fallback to local management if Redis fails
+        SESSION_POOL = {}
+        USER_POOL = {}
+        USAGE_POOL = {}
+        acquire_func = release_func = renew_func = lambda: True
```

Comment:
We might want to exit the application if this happens since, with replication, the app would behave oddly.